src/backend/distributed/master/master_repair_shards.c

```diff
@@ -297,7 +296,8 @@ CopyShardPlacement(int64 shardId, char *sourceNodeName, int32 sourceNodePort,
 
 	HOLD_INTERRUPTS();
 
-	SendCommandListInSingleTransaction(targetNodeName, targetNodePort, ddlCommandList);
+	SendCommandListToWorkerInSingleTransaction(targetNodeName, targetNodePort, nodeUser,
```

Comment:
> Dropping the placement as the as extension owner should be ok, so you can use worker_transactions for that.

We currently not use worker transactions for drop. We'll check-in drop on the enterprise repo once worker transactions are checked-in. And, if I understand correctly, we could use extension owner.

> Creating the shard placement should not use 2PC.

So, we should use `ExecuteRemoteCommand()` in a loop, and on any failure, fail the current transaction as well? Is that the expected behaviour?